src/jvm/main/org/jetbrains/kotlinx/lincheck/util/Types.kt

```diff
@@ -16,4 +16,43 @@ import sun.nio.ch.lincheck.Types.*
 import sun.nio.ch.lincheck.Types.convertAsmMethodType
 
 internal fun Method.toMethodSignature() = MethodSignature(this.name, convertAsmMethodType(this.descriptor))
-internal fun java.lang.reflect.Method.toMethodSignature() = Method.getMethod(this).toMethodSignature()
\ No newline at end of file
+internal fun java.lang.reflect.Method.toMethodSignature() = Method.getMethod(this).toMethodSignature()
+
+internal data class MethodDescriptor(
+    val className: String,
+    val methodSignature: MethodSignature
+) {
+    constructor(className: String, methodName: String, desc: String) :
+        this(className, MethodSignature(methodName, convertAsmMethodType(desc)))
+
+    val methodName: String get() = methodSignature.name
+    val returnType: Type get() = methodSignature.methodType.returnType
+    val argumentTypes: List<Type> get() = methodSignature.methodType.argumentTypes
+}
```

Comment:
Please add `toString()` override for `MethodDescriptor` (after rebase you'll see other classes also have this override #588).

Reply:
added